npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -166,7 +166,7 @@ def build_dependency(file:, type:, name:, requirement:)
 
       sig { override.void }
       def check_required_files
-        raise "No package.json!" unless get_original_file("package.json")
+        raise DependencyFileNotFound.new(nil, "package.json not found.") unless get_original_file("package.json")
```

Comment:
can we please add unit tests to capture this failure mode?

Reply:
@abdulapopoola , added test cases